dleq/cgo_dleq_test.go

```diff
@@ -25,3 +30,39 @@ func TestCGODLEq(t *testing.T) {
 	ed25519Pub := sk.Public().Bytes()
 	require.Equal(t, res.ed25519Pub[:], ed25519Pub)
 }
+
+func TestProofSecretComputesVerifyPubKeys(t *testing.T) {
+	// It would be nice to increase the number of iterations, but it's pretty slow even at 128. We
+	// previously had an issue when X or Y needed at least one high order padding byte. The chance
+	// of that happening is around (1/256+1/256)=1/128, so this loop will see values like that
+	// frequently, even if it doesn't happen on every run.
+	const iterations = 128
+	for i := 0; i < iterations; i++ {
+		proof, err := (&CGODLEq{}).Prove()
+		require.NoError(t, err)
+		res, err := (&CGODLEq{}).Verify(proof)
+		require.NoError(t, err)
+
+		// The ETH library needs the secret in big-endian format, while the monero library wants it
+		// in little endian format.
+		secretLE := proof.secret[:]
+		secretBE := common.Reverse(secretLE)
```

Comment:
This is where I got burned by the original behavior of Reverse.  :) The backing array is in the proof struct, so this was changing the order for all 3 variables.